crates/codegen/parser/runtime/src/query_parser_tests.rs

```diff
@@ -0,0 +1,26 @@
+use super::query_model::Query;
+
+fn run_parser_test(input: &str, result: &str) {
+    assert_eq!(Query::parse(input).unwrap().unparse(), result);
+}
+
+#[test]
+fn test_parsing() {
+    run_parser_test(
+        r#"@root [_ ... [Identifier] ["abc\\\"\n\r\b\t\u{01abcd}"] ...]"#,
```

Comment:
I suggest adding a separate test for each query/variant .. to test it e2e.

Reply:
I think a few edge case tests that show how the input is parsed and then unparsed to a canonical structure is all that is needed, given the engine takes query strings as test inputs and should be more thorough.

Reply:
Done

Reply:
at least a simple basic test for each syntax feature/variant we have, to protect against accidental regressions/changes in the future.